plugins/optimization-detective/storage/rest-api.php

```diff
@@ -88,6 +88,42 @@ function od_register_endpoint(): void {
 }
 add_action( 'rest_api_init', 'od_register_endpoint' );
 
+/**
+ * Determines if the HTTP origin is an authorized one.
+ *
+ * Note that `is_allowed_http_origin()` is not used directly because the underlying `get_allowed_http_origins()` does
+ * not account for the URL port (although there is a to-do comment committed in core to address this). Additionally,
+ * the `is_allowed_http_origin()` function in core for some reason returns a string rather than a boolean.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @see get_allowed_http_origins()
+ * @see is_allowed_http_origin()
+ *
+ * @param string $origin Origin to check.
+ * @return bool Whether the origin is allowed.
+ */
+function od_is_allowed_http_origin( string $origin ): bool {
```

Comment:
Why can't we use `is_allowed_http_origin()` directly instead of this function?

If it's just about the port support, I'd say that's an edge case so then I think we should at least use the original function anytime there's no port, and the custom logic only if there is one.

I'd also say we could simplify it: If there is a port in the passed `$origin`, simply remove it for the comparison and call `is_allowed_http_origin()` with that. It's not really that important to validate that the port also matches IMO, especially because of how much of an edge case that is.

Reply:
Well, unit tests run the home being `localhost:8889` so it's not an edge case from that perspective. 

I've gone ahead and just stripped out the port number in 545f0c3e6.